### ISSUE TITLE
Brains have a very limited feature set.

### ISSUE THREAD

**Issue by Alexei-B** [Alexei-B/Maze-Machine-Learning-Demo#7]
Brains need to have a more complete feature set to be practical as a neural net implementation. In particular adding new activation methods such as relu, softmax, softplus, tanh, and so on, as well as customising the layer architecture of the network, would greatly expand the use cases of the code.
Also they should be renamed "NNet" because "Brain" is a bit of an exaggeration.
